tests/test_optimizely.py

```diff
@@ -3911,6 +3912,39 @@ def test_get_feature_variable_returns__default_value__complex_audience_match(sel
         self.assertEqual(10, opt_obj.get_feature_variable_integer('feat2_with_var', 'z', 'user1', {}))
         self.assertEqual(10, opt_obj.get_feature_variable('feat2_with_var', 'z', 'user1', {}))
 
+    def test_get_optimizely_config__invalid_object(self):
+        """ Test that get_optimizely_config logs error if Optimizely instance is invalid. """
+
+        class InvalidConfigManager(object):
+            pass
+
+        opt_obj = optimizely.Optimizely(json.dumps(self.config_dict), config_manager=InvalidConfigManager())
+
+        with mock.patch.object(opt_obj, 'logger') as mock_client_logging:
+            self.assertIsNone(opt_obj.get_optimizely_config())
+
+        mock_client_logging.error.assert_called_once_with(
+            'Optimizely instance is not valid. Failing "get_optimizely_config".')
+
+    def test_get_optimizely_config__invalid_config(self):
+        """ Test that get_optimizely_config logs error if config is invalid. """
+
+        opt_obj = optimizely.Optimizely('invalid_datafile')
+
+        with mock.patch.object(opt_obj, 'logger') as mock_client_logging:
+            self.assertIsNone(opt_obj.get_optimizely_config())
+
+        mock_client_logging.error.assert_called_once_with(
+            'Invalid config. Optimizely instance is not valid. ' 'Failing "get_optimizely_config".'
+        )
+
+    def test_get_optimizely_config_returns_instance_of_optimizely_config(self):
+        """ Test that get_optimizely_config returns an instance of OptimizelyConfig. """
+
+        opt_obj = optimizely.Optimizely(json.dumps(self.config_dict_with_features))
+        opt_config = opt_obj.get_optimizely_config()
+        self.assertIsInstance(opt_config, optimizely_config.OptimizelyConfig)
+
```

Comment:
Can we add tests for OptConfig contents validation as well?

Reply:
Just pushed.